app/javascript/pages/country/widgets/widget-tree-cover-gain/widget-tree-cover-gain.js

```diff
@@ -16,57 +13,91 @@ export { default as actions } from './widget-tree-cover-gain-actions';
 const mapStateToProps = state => ({
   location: state.location.payload,
   isLoading: state.widgetTreeCoverGain.isLoading,
-  totalAmount: state.widgetTreeCoverGain.totalAmount,
-  percentage: state.widgetTreeCoverGain.percentage,
-  settings: state.widgetTreeCoverGain.settings,
-  locations: state.widgetTreeCoverGain.locations
+  gain: state.widgetTreeCoverGain.gain,
+  treeExtent: state.widgetTreeCoverGain.treeExtent,
+  indicators: state.widgetTreeCoverGain.indicators,
+  settings: state.widgetTreeCoverGain.settings
 });
 
-const WidgetTreeCoverGainContainer = props => {
-  const setInitialData = newProps => {
-    setWidgetData(newProps);
+class WidgetTreeCoverGainContainer extends PureComponent {
+  componentWillUpdate(nextProps) {
+    const { isMetaLoading, settings } = this.props;
+
+    if (JSON.stringify(settings) !== JSON.stringify(nextProps.settings)) {
```

Comment:
You can also use lodash 'isEqual' which performs a deep compare of arrays and objects. I have not preferences, but is faster than stringifying for large objects.